modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -2541,6 +2541,8 @@ void TFImporter::parsePack(tensorflow::GraphDef& net, const tensorflow::NodeDef&
     if (dim != 0)
         CV_Error(Error::StsNotImplemented, "Unsupported mode of pack operation.");
 
+    data_layouts[name] = DATA_LAYOUT_UNKNOWN;
```

Comment:
In `tf_importer`, some layers are sensitive to the input's data layout (like `Transpose`). And these layers may error if they are after the `Pack` layer.
Maybe we can judge the output data layout by the input data layout. 
For example, the input is 4 dim and NHWC data layout, which means the output should be the 5 dims.
But I'm not sure if this is the right way to do it

Reply:
We can consider cases that have defined data layout in `tf_importer`, for example stacking hwc can have nhwc data layout output. The problem is we cannot take every single data layout into account since theoratically there is countless.

I wonder in which case pack is followed by transpose though.

Reply:
I think we keep it unknown for the output data layout of pack since there are only 4 data layouts defined:

```cpp
enum DataLayout
{
    DATA_LAYOUT_NHWC,
    DATA_LAYOUT_NCHW,
    DATA_LAYOUT_NDHWC,
    DATA_LAYOUT_UNKNOWN,
    DATA_LAYOUT_PLANAR  // 2-dimensional outputs (matmul, flatten, reshape to 2d)
};
```

These data layouts do not satisfy our need if we want to describe the output data layout of pack unless more are added.

Reply:
A complete test example would be helpful.
From [stack API](https://www.tensorflow.org/api_docs/python/tf/stack), we can find that the axis is `[-(R+1), R+1)`.
The following code generates all possible cases of `tf.stack`.
```
for d in range(1, 7):
    for axis in range(-d - 1, d + 1):
        inp0 = tf.placeholder(tf.float32, list(range(1, d + 1)), 'input0')
        constant = tf.random_normal(list(range(1, d + 1)))
        pack_model = tf.stack([inp0, constant])
        print(pack_model)
        # save(inp0, pack_model, f'pack_dim{d}_{axis}', optimize=False)
```

Reply:
Testing a single `Pack` operator does not make any sense in this pull request. This pull request is changing the output data layout, and with or without this change, such tests will always pass.